src/omnicore/notifications.cpp

```diff
@@ -19,8 +20,41 @@
 namespace mastercore
 {
 
-//! Stores alert messages for Omni Core
-static std::string global_alert_message;
+//! Vector of currently active Omni alerts
+std::vector<AlertData> currentOmniAlerts;
+
+/**
+ * Adds a new alert to the alerts vector
+ *
+ */
+void AddAlert(uint16_t alertType, uint32_t alertExpiry, const std::string& alertMessage)
+{
+    AlertData newAlert;
+    newAlert.alert_type = alertType;
+    newAlert.alert_expiry = alertExpiry;
+    newAlert.alert_message = alertMessage;
+
+    // very basic sanity checks to catch malformed packets
```

Comment:
Alright, those are good points.  

I wanted some basic sanity checks in place because I hadn't built a way to clear a mistakenly created alert (eg if a blocktime was used but the alert type was set to block that would be a never expiring alert).

What I think I'll do is add a 'clear all' alert type that removes alerts, but restrict it to just the senders alerts (so that for example I couldn't clear an alert that you/the foundation had issued, but I could clear my own alerts).

With a way to clear erroneous alerts the sanity checks then become less important.